typing/printpat.ml

```diff
@@ -56,17 +56,21 @@ let rec pretty_val : type k . _ -> k general_pattern -> _ = fun ppf v ->
   | Tpat_constant c -> fprintf ppf "%s" (pretty_const c)
   | Tpat_tuple vs ->
       fprintf ppf "@[(%a)@]" (pretty_vals ",") vs
-  | Tpat_construct (_, cstr, []) ->
+  | Tpat_construct (_, cstr, [], _) ->
```

Comment:
Should maybe also be `None` in the new argument? Probably can't happen, but more obviously correct.

Reply:
Not really more correct: it would result in printing `C (type a) ( : _)`.
Or do you mean that we should print something clearly wrong if the invariant is broken?

Reply:
Sorry, I was confusing this bit of code with the stuff below, and thought it was related to handling `::`.